mindsdb/integrations/handlers/influxdb_handler/influxdb_tables.py

```diff
@@ -27,68 +29,52 @@ def select(self, query: ast.Select) -> pd.DataFrame:
-        if query.order_by and len(query.order_by) > 0:
-            order_by_conditions["columns"] = []
-            order_by_conditions["ascending"] = []
-
-            for an_order in query.order_by:
-                if an_order.field.parts[0] != "":
-                    next    
-                if an_order.field.parts[1] in self.get_columns():
-                    order_by_conditions["columns"].append(an_order.field.parts[1])
-
-                    if an_order.direction == "ASC":
-                        order_by_conditions["ascending"].append(True)
-                    else:
-                        order_by_conditions["ascending"].append(False)
-                else:
-                    raise ValueError(
-                        f"Order by unknown column {an_order.field.parts[1]}"
-                    )
-
-        influxdb_tables_df = self.handler.call_influxdb_tables()
-
-        selected_columns = []
-        for target in query.targets:
-            if isinstance(target, ast.Star):
-                selected_columns = self.get_columns()
-                break
-            elif isinstance(target, ast.Identifier):
-                selected_columns.append(target.parts[-1])
-            else:
-                raise ValueError(f"Unknown query target {type(target)}")
-
-
-        if len(influxdb_tables_df) == 0:
-            influxdb_tables_df = pd.DataFrame([], columns=selected_columns)
-        else:
-            influxdb_tables_df.columns = self.get_columns()
-            for col in set(influxdb_tables_df.columns).difference(set(selected_columns)):
-                influxdb_tables_df = influxdb_tables_df.drop(col, axis=1)
-
-            if len(order_by_conditions.get("columns", [])) > 0:
-                influxdb_tables_df = influxdb_tables_df.sort_values(
-                    by=order_by_conditions["columns"],
-                    ascending=order_by_conditions["ascending"],
-                )
-
-        if query.limit:
-            influxdb_tables_df = influxdb_tables_df.head(query.limit.value)
+        
+        table_name=self.handler.connection_data['influxdb_table_name']
+        head_influx_table = self.handler.call_influxdb_tables(f"SELECT * FROM {table_name} LIMIT 1")
+        select_statement_parser = SELECTQueryParser(
+            query,
+            "tables",
+            self.get_columns(head_influx_table)
+        )
+        selected_columns, where_conditions, order_by_conditions, _ = select_statement_parser.parse_query()
+        
+        try:
+            selected_columns.remove("name")
+            selected_columns.remove("tags")
+        except Exception as e:
+            logger.warn(e)
+
+        formatted_query=self.get_select_query(table_name,selected_columns,where_conditions,order_by_conditions,query.limit)
+        influxdb_tables_df  = self.handler.call_influxdb_tables(formatted_query)
 
         return influxdb_tables_df
 
-    def get_columns(self) -> List[str]:
+    def get_columns(self,dataframe) -> List[str]:
```

Comment:
Hey @Tanmai2002,
If you take a look at the `APITable` interface, `get_columns()` is an abstract method that needs to be implemented. Therefore, let's not change the method signature and keep in the same. Since this table is fixed, I think you can move your logic of getting the columns from the head of the `DataFrame` directly to this method without requiring a `dataframe` parameter. I mean this,
`head_influx_table = self.handler.call_influxdb_tables(f"SELECT * FROM {table_name} LIMIT 1")`